[JuliaLowering] Reconstruct macro name before, not during, macro expansion

mlechu opened the pull request:
On top of https://github.com/JuliaLang/julia/pull/60162.  This simplification lets macro expansion handle the macro name as an identifier rather than the nonterminal `K"macro_name"` node.  That and the old `MacroName` node (as well as `CmdMacroName`, `StrMacroName`) are mainly the result of restrictions on raw parser output: we can't parse `@Base.x` to a non-contiguous `@x`, even though that's what we want in the AST in almost every case.

topolarity commented on JuliaLowering/src/compat.jl:
```diff
@@ -277,14 +277,10 @@ function _insert_convert_expr(@nospecialize(e), graph::SyntaxGraph, src::SourceA
-                child_exprs[1] = a1_esc(Expr(:(.), a1.args[1],
-                                             Expr(:macro_name, a12_esc(a12.value))))
-            end
+        if a1 isa Symbol && a1 === :@__dot__
```
```julia
julia> typeof(:@__dot__)
Expr
```

Probably needs some test coverage

topolarity replied:
```suggestion
        if a1 isa Symbol && a1 === Symbol("@__dot__")

mlechu committed:
Reconstruct macro name before, not during, macro expansion

mlechu committed:
fixes

Co-authored-by: Cody Tapscott <topolarity@tapscott.me>

mlechu committed:
Update list of precompilable stdlibs

- StyledStrings, Markdown, JuliaSyntaxHighlighting, and Profile now pass
- Include REPLExt and SparseArraysExt

mlechu committed:
add test

topolarity reviewed (approved):
Another nice simplification. Thanks, @mlechu !
